spec/defines/versionlock_spec.rb

```diff
@@ -65,15 +85,30 @@
           )
         end
       end
+    end
+  end
+
+  context 'on Suse 15' do
+    test_on = {
+      hardwaremodels: ['x86_64'],
+      supported_os: [
+        {
+          'operatingsystem' => 'Suse',
```

Comment:
This will limit the tests to Suse 15 (where we currently don't have any factsets). The tests won't be executed for SLES 15 or any future releases. An alternative approach would be:

```diff
diff --git a/spec/defines/versionlock_spec.rb b/spec/defines/versionlock_spec.rb
index 1480802..f28a761 100644
--- a/spec/defines/versionlock_spec.rb
+++ b/spec/defines/versionlock_spec.rb
@@ -85,29 +85,14 @@ describe 'zypprepo::versionlock' do
           )
         end
       end
-    end
-  end
-
-  context 'on Suse 15' do
-    test_on = {
-      hardwaremodels: ['x86_64'],
-      supported_os: [
-        {
-          'operatingsystem' => 'Suse',
-          'operatingsystemrelease' => ['15'],
-        },
-      ],
-    }
-    on_supported_os(test_on).each do |_os, os_facts|
-      let(:facts) { os_facts }

-      context 'with an invalid title' do
+      context 'with an invalid title', if: os_facts[:os]['release']['major'].to_i >= 15 do
         let(:title) { titles['invalid'] }

         it { is_expected.to raise_error(Puppet::PreformattedError, %r(%\{NAME\}-%\{VERSION\}-%\{RELEASE\}\.%\{ARCH\})) }
       end

-      context 'with an invalid wildcard pattern' do
+      context 'with an invalid wildcard pattern', if: os_facts[:os]['release']['major'].to_i >= 15 do
         let(:title) { titles['invalid_wildcard'] }

         it { is_expected.to raise_error(Puppet::PreformattedError, %r(%\{NAME\}-%\{VERSION\}-%\{RELEASE\}\.%\{ARCH\})) }
```

Reply:
Personally I prefer adding SLes15 facts to facterdb.

Reply:
https://github.com/voxpupuli/facterdb/pull/240

Reply:
latest facterdb release contains the facts for SLES/opensuse 15.